.github/workflows/build-apk.yml

```diff
@@ -0,0 +1,69 @@
+name: Build APK
+
+on:
+  push:
+    branches:
+      - main
+      - feature/set-up-github-APK-button
+  pull_request:
+    branches:
+      - main
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      # Checkout the latest code
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      # Set up JDK 11 for Android build
+      - name: Setup JDK
+        uses: actions/setup-java@v4
+        with:
+          distribution: "temurin"
+          java-version: "17"
+
+      # Cache Gradle dependencies
+      - name: Cache Gradle packages
+        uses: actions/cache@v3
```

Comment:
Why not use cache@v4?

Reply:
I will change all the version to v4 and see if it's still working,  for the on:push, I will delet that before merging. For some reason the CI don't pass, so I will try to make it work first.

Reply:
All versions had been changed to v4, I modified the "on:" to only activate on workflow_dispatch, everything work, ready to be merge, I ask for a re-review to make sure all change is beining aprroved.